picard/ui/coverartbox.py

```diff
@@ -169,16 +177,29 @@ def fetch_remote_image(self, url):
                     data = f.read()
                 self.load_remote_image(url, mime, data)
 
-    def on_remote_image_fetched(self, url, data, reply, error):
+    def on_remote_image_fetched(self, url, data, reply, error, fallbackData=None):
         mime = reply.header(QtNetwork.QNetworkRequest.ContentTypeHeader)
         if mime in ('image/jpeg', 'image/png'):
             self.load_remote_image(url, mime, data)
-        elif reply.url().hasQueryItem("imgurl"):
+        elif url.hasQueryItem("imgurl"):
             # This may be a google images result, try to get the URL which is encoded in the query
-            url = QtCore.QUrl(reply.url().queryItemValue("imgurl"))
+            url = QtCore.QUrl(url.queryItemValue("imgurl"))
             self.fetch_remote_image(url)
         else:
-            log.warning("Can't load image with MIME-Type %s", mime)
+            log.warning("Can't load remote image with MIME-Type %s", mime)
+            if fallbackData:
+               # Tests for image format obtained from file-magic
+               if fallbackData[:2]=='\xff\xd8':
```

Comment:
Sorry, didn't notice.

https://github.com/metabrainz/picard/blob/master/picard/util/imageinfo.py#L72 
Looking at above, since there seems to be a precedent of using [:n], keep it as is.